scripts/labdeploy.ps1

```diff
@@ -912,9 +915,88 @@ if ($setupNewVC) {
+        $ovfconfig.Common.instance_id.value = New-Guid
+        $ovfconfig.Common.password.value = $RouterVMPassword
+        $ovfconfig.Common.user_data.value = [Convert]::ToBase64String($routerUserDataBytes)
+
+        Write-Log "Deploying Routing VM $RouterVMDisplayName ..."
+        $vm = Import-VApp -Server $vc -Source $RouterOVA -OvfConfiguration $ovfconfig -Name $RouterVMHostname -VMHost $VMhost -Datastore $vcdatastore -DiskStorageFormat thin -Force
+
+        Write-Log "Attaching Routing VM $RouterVMDisplayName to workload segment..."
+        New-NetworkAdapter -VM $vm -NetworkName $NewVCWorkloadDVPGName -StartConnected | Out-Null
+
+        Write-Log "Powering On $RouterVMDisplayName ..."
+        $vm | Start-Vm | Out-Null # wait for tools
+
+        # Create static route on NSX segment to reach VM segment
+        # Connecting to NSX-T Manager
+        Write-Log "Connecting to NSX-T Server $nsxtHost ..."
+        $nsxtConnection = Connect-NsxServer -Server ${nsxtHost} -User ${nsxtUser} -Password ${nsxtPass}
+        Write-Log "Connected to NSX-T Server"
+        # Get Default T1 Gateway
+        Write-Log "Getting NSX-T Default T1 Gateway"
+        $t1result = (Invoke-ListTier1 -Server $nsxtConnection).results | Where-Object { $_.DisplayName -like 'TNT**-T1' }
+        $t1Name = $t1result.DisplayName
+
+        $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes("$($nsxtUser):$($nsxtPass)"))
+        $Header = @{
+            Authorization = "Basic $base64AuthInfo"
+        }
+    
+        # Remove existing route with same name
+        $existingRoute = (Invoke-ListTier1StaticRoutes -Server $nsxtConnection -Tier1Id $t1Name).Results | Where-Object { 
+            $_.network -eq "10.${groupNumber}.1${labNumber}.128/27"
+        }
+        if ($existingRoute) {
+            Write-Log "Removing an exisitng route with same network target"
+            $sRouteURL = "https://$nsxtHost/policy/api/v1/infra/tier-1s/$t1Name/static-routes/" + $existingRoute.Id
+            Invoke-RestMethod -Uri $sRouteURL -Headers $Header -Method DELETE -ContentType "application/json" -SkipCertificateCheck
+            Start-Sleep 15
+        }
+        
+        $Body = @"
+        {
+            "display_name": "$NewVcVAppName",
+            "network": "10.${groupNumber}.1${labNumber}.128/27",
+            "next_hops":[
+                {
+                    "admin_distance":1,
+                    "ip_address": "$RouterVMIPAddress"
+                }
+            ],
+            "id":"$NewVcVAppName"
         }
-        Write-Log "Workload VMs deployed successfully ..."
+"@
+        $sRouteURL = "https://$nsxtHost/policy/api/v1/infra/tier-1s/$t1Name/static-routes/$NewVcVAppName"
+        $sRoute = Invoke-RestMethod -Uri $sRouteURL -Headers $Header -Method PUT -Body $Body -ContentType "application/json" -SkipCertificateCheck
+        Start-Sleep 15
+        Write-Log "Static route $NewVcVAppName created....."
     }
 
```

Comment:
@lrivallain . I cannot find line 1004. But yes. toward the end. We can check if the connection is still open, then we should close it to avoid exhausting resources. Also same for NSX-T please.